pkg/docker/config/config.go

```diff
@@ -284,11 +284,11 @@ func getPathToAuth(sys *types.SystemContext) (string, bool, error) {
 // if the file exists and is empty, readJSONFile returns an error
 func readJSONFile(path string, legacyFormat bool) (dockerConfigFile, error) {
 	var auths dockerConfigFile
+	auths.AuthConfigs = map[string]dockerAuthConfig{}
```

Comment:
Is it possible for AuthConfigs to have a valid value at this point?  If so, I don't think we want to reset it to empty here.

Reply:
As far as I know, we just declear `auths` at line#286 and assign value for it  at line#304
`if err = json.Unmarshal(raw, &auths)`.  So there is no valid value at this point :-)